presto-iceberg/src/main/java/com/facebook/presto/iceberg/function/changelog/ApplyChangelogFunction.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.function.changelog;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.BlockIndex;
+import com.facebook.presto.spi.function.BlockPosition;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import io.airlift.slice.Slice;
+import org.apache.iceberg.ChangelogOperation;
+
+import java.util.Optional;
+
+@AggregationFunction("apply_changelog")
```

Comment:
Should these functions be hidden?  Do we expect people to use them directly?

Reply:
I'm not exactly clear on what the difference between hidden and non-hidden functions are. I couldn't find it in the docs.

To give you an idea on the use case for this function, there will eventually be some system (may be internal or external to Presto) which runs a query like the following:

```SQL
SELECT 
    rowdata."${primary_key}", 
    apply_changelog(ordinal, operation, rowdata order by ordinal) as rowdata
FROM "${tableName}$changelog"
GROUP BY rowdata."${primary_key}"
```

Most of the time, users probably won't be running this directly but whatever application that performs sample maintenance will need to have access to the function

Reply:
Hidden functions don't show up in `SHOW FUNCTIONS`, and we may deliberately leave them undocumented, but they are otherwise accessible.  Seems to fit this use case.